SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp

```diff
@@ -687,18 +689,21 @@ void USpatialReceiver::HandleActorAuthority(const Worker_AuthorityChangeOp& Op)
 						UpdateShadowData(Op.entity_id);
 					}
 
-					Actor->OnAuthorityGained();
+					if (bActorHadAuthority) // This is Actor's spawning worker.
```

Comment:
Could we extend the comment here to explain why this is the case? ie. the *only* time a worker should already have Unreal authority when receiving an spatial authority gained event is if this worker spawned it. 

Also thinking this is a little implementation dependent, is there no other way we could determine if we created this entity? SpatialReceiver has `PendingActorRequests` to track creation calls, but I'm not sure it's much better. Track all entities created by this worker? We could clear it when we lose authority also to prevent it becoming unbounded.

Reply:
Currently, it seems like `PendingActorRequests` is purely used to handle the lifecycle of entity creation requests (and those requests are removed when the `CreateEntityResponse` op is received).

While I agree to make this implementation-dependent on our handling of things like authority, which have been changed recently, isn't ideal - I think it's the best thing we currently have for determining the locally spawned Actor state (vs. the two others: previously unknown but now authoritative Actor, and previously interested but now authoritative Actor). The latter case makes it hard to determine if something was spawned locally just by whether a new Actor was created for example. 

I think creating a new collection alongside `PendingActorRequests` to track a similar thing seems like too much overhead for concern over whether an invariant (that we rely on in a bunch of other places) might change? It also shouldn't be hard to change in the future if worse comes to worst?

Reply:
This is probably the third of fourth feature recently that has warranted a collection tracking entities this worker created up to the point where authority is gained over them so I think we should look into making that.
The main question is should it be in this PR or another one. 

I'd argue for it here because, as Mike points out, whenever we change how we handle roles we end up missing things like this.
I could see this possibly breaking if the worker gives up authority (due to lb) then get's authority back very shortly afterwards. (I think this doesn't actually break right now but we depend on the fine details of ops to for that).

I agree with Ally that we can't use anything relating to PendingActorRequests as that will introduce bugs.

Reply:
If there are 3 or 4 other features in the GDK will a similar dependency, then I agree reconciling those uses into something specifically suited to tracking Actor lifecycles is better. I created a ticket for tracking [here](https://improbableio.atlassian.net/browse/UNR-3960).

However, I don't think here is the ideal case to do that. 1) this is a task that needs to go in this sprint given NWX continual pursuit of its completeness and I'm conscious of broadening scope (even in implementation details) 2) I'd generally prefer to separate out this PR relating purely to changing lifecycle event semantics and a PR which aims to broadly reconcile functionality across multiple other systems.

Reply:
I'm okay with a follow-up, we should prioritise. Could you also tag the ticket number in the comment where `bActorHadAuthority` is set.

Reply:
I'm okay with a follow-up, we should prioritise. Could you also tag the ticket number in the comment where `bActorHadAuthority` is set.